internal/application/application.go

```diff
@@ -131,44 +139,53 @@ func (a *Application) Bind(service interfaces.Service) error {
 		return err
 	}
 
-	deployment, err := a.deployment()
-	if err != nil {
-		return err
-	}
+	for {
+		deployment, err := a.deployment()
+		if err != nil {
+			return err
+		}
 
-	volumes := deployment.Spec.Template.Spec.Volumes
+		volumes := deployment.Spec.Template.Spec.Volumes
 
-	for _, volume := range volumes {
-		if volume.Name == service.Name() {
-			return errors.New("service already bound")
+		for _, volume := range volumes {
+			if volume.Name == service.Name() {
+				return errors.New("service already bound")
+			}
 		}
-	}
 
-	volumes = append(volumes, corev1.Volume{
-		Name: service.Name(),
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName: bindSecret.Name,
+		volumes = append(volumes, corev1.Volume{
+			Name: service.Name(),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: bindSecret.Name,
+				},
 			},
-		},
-	})
-	// TODO: Iterate over containers and find the one matching the app name
-	volumeMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
-	volumeMounts = append(volumeMounts, corev1.VolumeMount{
-		Name:      service.Name(),
-		ReadOnly:  true,
-		MountPath: fmt.Sprintf("/services/%s", service.Name()),
-	})
-
-	deployment.Spec.Template.Spec.Volumes = volumes
-	deployment.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
-
-	_, err = a.kubeClient.Kubectl.AppsV1().Deployments(deployments.WorkloadsDeploymentID).Update(
-		context.Background(),
-		deployment,
-		metav1.UpdateOptions{},
-	)
-	return err
+		})
+		// TODO: Iterate over containers and find the one matching the app name
+		volumeMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      service.Name(),
+			ReadOnly:  true,
+			MountPath: fmt.Sprintf("/services/%s", service.Name()),
+		})
+
+		deployment.Spec.Template.Spec.Volumes = volumes
+		deployment.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
+
+		_, err = a.kubeClient.Kubectl.AppsV1().Deployments(deployments.WorkloadsDeploymentID).Update(
+			context.Background(),
+			deployment,
+			metav1.UpdateOptions{},
+		)
+
+		if err == nil || !apierrors.IsConflict(err) {
```

Comment:
This is essentially the same exit criteria as in lines 113 - 118 above. I would use the same code (two separate `if`` statements) at both places

Reply:
Ok. A moment ...

Reply:
Addressed with commit 7ae5bb8452eda05992bb49eb97d524efdb6b8fd1